core/bloombits/utils.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+package bloombits
+
+import (
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type (
+	BitVector  []byte
+	CompVector []byte
+)
+
+// bvAnd binary ANDs b to a
+func bvAnd(a, b BitVector) {
+	for i, bb := range b {
+		a[i] &= bb
+	}
+}
+
+// bvOr binary ORs b to a
+func bvOr(a, b BitVector) {
+	for i, bb := range b {
+		a[i] |= bb
+	}
+}
+
+// bvZero returns an all-zero bit vector
+func bvZero(sectionSize int) BitVector {
```

Comment:
I really don't like having an extra method for this. Having an explicit `make` in a code makes it more readable, imho we shouldn't hide it just to make it fancy. It's more explicit that way imho, though I'm up for debate.